conda-store-server/tests/_internal/alembic/version/test_89637f546129_remove_conda_package_build_channel.py

```diff
@@ -0,0 +1,183 @@
+                "subdir": "linux-64",
+            },
+            {
+                "id": 2,
+                "build": "py311h06a4308_0",
+                "package_id": 1,
+                "build_number": 0,
+                "sha256": "two",
+                "subdir": "linux-64",
+            },
+            {
+                "id": 3,
+                "build": "py38h06a4308_0",
+                "package_id": 1,
+                "build_number": 0,
+                "sha256": "three",
+                "subdir": "linux-64",
+            },
+            {
+                "id": 4,
+                "build": "py39h06a4308_0",
+                "package_id": 2,
+                "build_number": 0,
+                "sha256": "four",
+                "subdir": "linux-64",
+            },
+            {
+                "id": 5,
+                "build": "py310h06a4308_0",
+                "package_id": 2,
+                "build_number": 0,
+                "sha256": "five",
+                "subdir": "linux-64",
+            },
+        ]
+        default_values = {
+            "depends": "",
+            "md5": "",
+            "timestamp": 0,
+            "constrains": "",
+            "size": 0,
+        }
+        for cpb in conda_package_builds:
+            conda_package = orm.CondaPackageBuild(**cpb, **default_values)
+            db.add(conda_package)
+        db.commit()
+
+        # force in some channel data
+        # conda_package_build 1 should have package_id 2 after migration
+        db.execute(text("UPDATE conda_package_build SET channel_id=2 WHERE id=1"))
+        # conda_package_build 2 should have package_id 1 after migration
+        db.execute(text("UPDATE conda_package_build SET channel_id=1 WHERE id=2"))
+        # conda_package_build 3 should have package_id 1 after migration
+        db.execute(text("UPDATE conda_package_build SET channel_id=1 WHERE id=3"))
+        # conda_package_build 4 should have package_id 2 after migration
+        db.execute(text("UPDATE conda_package_build SET channel_id=2 WHERE id=4"))
+
+        # don't set conda_package_build 5 channel_id as a test case
+        # conda_package_build 5 package_id should be unchanged (2) after migration
+
+        db.commit()
+
+
+def test_remove_conda_package_build_channel_basic(
+    conda_store, alembic_config, alembic_engine, alembic_runner
+):
+    """Simply run the upgrade and downgrade for this migration"""
+    # migrate all the way to the target revision
+    alembic_runner.migrate_up_to("89637f546129")
+
+    # try downgrading
+    alembic_runner.migrate_down_one()
+
+    # try upgrading once more
+    alembic_runner.migrate_up_one()
+
+
+def test_remove_conda_package_build_bad_data(
+    conda_store, alembic_config, alembic_engine, alembic_runner
+):
+    """Simply run the upgrade and downgrade for this migration"""
+    # migrate all the way to the target revision
+    alembic_runner.migrate_up_to("89637f546129")
+
+    # try downgrading
```

Comment:
Since the goal of the migration to delete the `channel_id` off the `conda_package_build` table, it might be good to assert that there's no `channel_id` column in that table here...

Reply:
good idea!